### ISSUE TITLE
`leanpkg build` fails with `env: illegal option -- u` on OS X

### ISSUE THREAD

**Issue by kim-em** [leanprover/lean3#1564]
The OS X version of [env](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/env.1.html) doesn't have an `-u` (`--unset`) option. I'm not sure if having LEAN_PATH set to the empty string would suffice: if so it seems calling `env LEAN_PATH= lean --make` would work everywhere.

In any case, I don't know why we need to [clear the LEAN_PATH environment variable](https://github.com/leanprover/lean/blob/master/leanpkg/leanpkg/main.lean#L48), so I'm not sure of the appropriate fix.

**Comment by gebner** [comment]
The reason why we clear the LEAN_PATH environment variable is because it would override the paths obtained from `leanpkg.path`.  (We set the LEAN_PATH environment variable in the leanpkg script so that Lean find the leanpkg files.)  In `leanpkg build`, we want to call `lean --make` with the path of the current package, not with the LEAN_PATH we inherit from leanpkg.

The plan here is to add `io.env.getenv`, `io.env.setenv`, and `io.env.unsetenv` calls, then we can do the environment modification inside leanpkg.

> if so it seems calling env LEAN_PATH= lean --make would work everywhere.

Unfortunately not.  An empty environment variable is still set, in this case Lean would then use the empty search path (that is, absolute `import` statements would not work at all):
```
[gebner:~/smt2_interface] master ± env LEAN_PATH= lean -p
{
  "is_user_leanpkg_path": false,
  "leanpkg_path_file": "/home/gebner/smt2_interface/leanpkg.path",
  "path": []
}
[gebner:~/smt2_interface] master ± env -u LEAN_PATH lean -p
{
  "is_user_leanpkg_path": false,
  "leanpkg_path_file": "/home/gebner/smt2_interface/leanpkg.path",
  "path": [
    "/home/gebner/lean/bin/../library",
    "/home/gebner/lean/bin/../lib/lean/library",
    "/home/gebner/smt2_interface/./."
  ]
}
```

**Comment by kim-em** [comment]
Interestingly, I just discovered the OSX 10.12's version of `env` supports the `-u` option, so `leanpkg` works fine there.

**Comment by gebner** [comment]
This should be fixed since 1b8533130b79fdb28e0210c11778aaa0f7eacee8.  Please reopen if the problem is still there.
